vulnerabilities/templates/vulnerability.html

```diff
@@ -20,33 +20,72 @@ <h3>Summary:</h3>
 
         {% if object_list %}
             <h3>References:</h3>
-                <table class="table is-bordered is-striped is-narrow is-hoverable is-fullwidth">
+            <table class="table is-bordered is-striped is-narrow is-hoverable is-fullwidth">
+                <tr>
+                    <th> Reference ID </th>
+                    <th> URL </th>
+                </tr>
+                {% for ref in object_list %}
                     <tr>
-                        <th> Reference ID </th>
-                        <th> URL </th>
+                        {% if ref.reference_id %}
+                            <td>{{ref.reference_id}}</td>
+
+                        {% else %}
+                            <td>-</td>
+
+                        {% endif %}
+
+                        {% if ref.url %}
+                            <td><a href="{{ ref.url }}">{{ref.url}}</a></td>
+
+                        {% else %}
+                            <td>-</td>
+
+                        {% endif %}
                     </tr>
-                {% for ref in object_list %}
+
+                {% endfor %}
+
+            </table>
+
+            <h3>Severity</h3>
+            <table class="table is-bordered is-striped is-narrow is-hoverable is-fullwidth">
+                <tr>
+                    <th> System </th>
+                    <th> Score </th>
+                    <th> Found At </th>
+                </tr>
+            {% for ref in object_list %}
+                {% for obj in ref.scores %}
                 <tr>
-                {% if ref.reference_id %}
-                    <td>{{ref.reference_id}}</td>
+                    {% if ref.scores %}
```

Comment:
IMHO ` if ref.scores ` is redundant since we are already iterating from it earlier.